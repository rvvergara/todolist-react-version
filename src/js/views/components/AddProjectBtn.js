import React from 'react';
import PropTypes from 'prop-types';

const AddProjectBtn = (props) => {
  const {
    clickAddProjectBtn,
  } = props;

  return (
    <button
      type="button"
      className="btn btn-primary mt-1"
      onClick={clickAddProjectBtn}
    >
    Add New Project
    </button>
  );
};

AddProjectBtn.propTypes = {
  clickAddProjectBtn: PropTypes.func.isRequired,
};

export default AddProjectBtn;
