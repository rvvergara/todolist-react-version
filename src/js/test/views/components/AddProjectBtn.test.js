import React from 'react';
import { shallow } from 'enzyme';
import AddProjectBtn from '../../../views/components/AddProjectBtn';

describe('AddProjectBtn', () => {
  let wrapper;
  let clickAddProjectBtn;

  beforeEach(() => {
    clickAddProjectBtn = jest.fn();

    wrapper = shallow(
      <AddProjectBtn
        addProjectMode={false}
        clickAddProjectBtn={clickAddProjectBtn}
      />,
    );
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call clickAddProjectBtn when button is clicked', () => {
    wrapper.find('button').simulate('click');
    expect(clickAddProjectBtn).toHaveBeenCalled();
  });
});
