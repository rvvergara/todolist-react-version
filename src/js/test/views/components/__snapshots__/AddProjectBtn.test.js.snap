// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddProjectBtn should render correctly 1`] = `
<button
  className="btn btn-primary mt-1"
  onClick={[MockFunction]}
  type="button"
>
  Add New Project
</button>
`;
